torch/distributed/tensor/_random.py

```diff
@@ -38,8 +40,8 @@ def is_rng_supported_mesh(device_mesh: DeviceMesh) -> bool:
         Currently we only support correct RNG on cuda/cuda-like devices.
     """
     device_handle = _get_device_handle(device_mesh.device_type)
-    if device_handle and hasattr(device_handle, "set_rng_state"):
-        return True
+    if device_handle and hasattr(device_handle, "rng_supported_mesh"):
+        return device_handle.rng_supported_mesh
```

Comment:
One question - so you would set `rng_supported_mesh` to `false` on your  device but you still need a DTensor RNG for your device, right? Then what's the difference in DTensor RNG bwt your own and pytorch? Is it possible to close the gap?

Reply:
As described in https://github.com/pytorch/pytorch/issues/138329, backends like CPU doesn't have complete support for DTensor RNG and gives a warning -

```
warnings.warn(
/usr/local/lib/python3.10/dist-packages/torch/distributed/tensor/_random.py:45: UserWarning: DTensor random operators may not have complete support on cpu device mesh
```

The above issue also mentions the following -

"...is there a plan to support the DTensor random ops for CPU better? It will not be offset based RNG tracker, but the seed can be advanced correctly for the CPU RNG state after the DTensor random op".

It would be good to know if this is in plan. However, as mentioned in https://github.com/pytorch/pytorch/pull/141141#issuecomment-2493182837, Gaudi is currently broken and fails with DTensor RNG ops as it uses CPU RNG state and has a set_rng_state() implemented. The purpose of this PR is to  stop the current failure on the Gaudi side and issue a warning similar to CPU for Gaudi. It would be welcome to close the gap as you mentioned to support DTensor RNG for CPU but probably can be taken up as a separate task?